src/keys/key-stretcher.js

```diff
@@ -37,72 +36,38 @@ module.exports = (cipherType, hash, secret, callback) => {
   const seed = Buffer.from('key expansion')
   const resultLength = 2 * (ivSize + cipherKeySize + hmacKeySize)
 
-  hmac.create(hash, secret, (err, m) => {
-    if (err) {
-      return callback(err)
-    }
-
-    m.digest(seed, (err, a) => {
-      if (err) {
-        return callback(err)
-      }
-
-      let result = []
-      let j = 0
-
-      whilst(
-        () => j < resultLength,
-        stretch,
-        finish
-      )
-
-      function stretch (cb) {
-        m.digest(Buffer.concat([a, seed]), (err, b) => {
-          if (err) {
-            return cb(err)
-          }
-
-          let todo = b.length
+  const m = await hmac.create(hash, secret)
+  let a = await m.digest(seed)
 
-          if (j + todo > resultLength) {
-            todo = resultLength - j
-          }
+  let result = []
+  let j = 0
 
-          result.push(b)
+  while (j < resultLength) {
+    const b = await m.digest(Buffer.concat([a, seed]))
+    let todo = b.length
 
-          j += todo
-
-          m.digest(a, (err, _a) => {
-            if (err) {
-              return cb(err)
-            }
-            a = _a
-            cb()
-          })
-        })
-      }
-
-      function finish (err) {
-        if (err) {
-          return callback(err)
-        }
+    if (j + todo > resultLength) {
```

Comment:
I'm not sure if this if block is necessary?

Reply:
Certainly removing it doesn't make any tests fail, though it's in the callback version too.